liquibase-standard/src/main/java/liquibase/database/core/DatabaseUtils.java

```diff
@@ -75,12 +75,53 @@ public static void initializeDatabase(String defaultCatalogName, String defaultS
                 }
                 executor.execute(new RawSqlStatement("USE " + schema));
             } else if (database instanceof MSSQLDatabase) {
-                    defaultCatalogName = StringUtil.trimToNull(defaultCatalogName);
-                    if (defaultCatalogName != null) {
-                        executor.execute(new RawSqlStatement(String.format("USE %s", defaultCatalogName)));
-                    }
+                defaultCatalogName = StringUtil.trimToNull(defaultCatalogName);
+                if (defaultCatalogName != null) {
+                    executor.execute(new RawSqlStatement(String.format("USE %s", defaultCatalogName)));
+                }
             }
         }
     }
 
+    public static void initializePostgresSearchPathPermanently(String defaultCatalogName, String defaultSchemaName, Database database) {
+
+        final Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc", database);
+
+        if (database instanceof PostgresDatabase && defaultSchemaName != null) {
+            String searchPath = null;
+            try {
+                searchPath = executor.queryForObject(new RawSqlStatement("SHOW SEARCH_PATH"), String.class);
+            } catch (Throwable e) {
+                Scope.getCurrentScope().getLog(DatabaseUtils.class).info("Cannot get search_path", e);
```

Comment:
Maybe this should be logged at a different level than info?

Reply:
Right, thinking of updating this as a warning and the other one you comment below as an error. Would that be the correct logging level for both of these messages?

Thanks,
Daniel.

Reply:
Sounds right to me